pyiceberg/table/upsert_util.py

```diff
@@ -86,7 +97,8 @@ def get_rows_to_update(source_table: pa.Table, target_table: pa.Table, join_cols
     if rows_to_update:
         rows_to_update_table = pa.concat_tables(rows_to_update)
     else:
-        rows_to_update_table = pa.Table.from_arrays([], names=source_table.column_names)
+        empty_arrays = [pa.array([], type=field.type) for field in source_table.schema]
+        rows_to_update_table = pa.Table.from_arrays(empty_arrays, schema=source_table.schema)
```

Comment:
Today I learned about the existence of this function:
```suggestion
        rows_to_update_table = source_table.schema.empty_table()
```
WDYT?

Reply:
This works for me .. Thanks for the suggestion @Fokko